File: frontend/src/types/food.ts
export interface Food {
  id: string
  userId: string
  name: string
  unit: string
  caloriesPerUnit: number
  proteinPerUnit: number
}

export interface CreateFoodRequest {
  name: string
  unit: string
  caloriesPerUnit: number
  proteinPerUnit: number
}

export interface UpdateFoodRequest {
  name: string
  unit: string
  caloriesPerUnit: number
  proteinPerUnit: number
}

File: backend/MealPrepper.Core/Entities/Food.cs
namespace MealPrepper.Core.Entities;

public class Food
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Unit { get; set; } = string.Empty;
    public decimal CaloriesPerUnit { get; set; }
    public decimal ProteinPerUnit { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public User User { get; set; } = null!;
    public ICollection<ShoppingListItem> ShoppingListItems { get; set; } = [];
}
